widgets/pulseaudio.lua

```diff
@@ -33,44 +33,49 @@ local function factory(args)
     --  '/device\\.string/ p' prints the line containing 'device.string:'.
     
     local settings    = args.settings or function()
+            widgettext  = volume_now.left .. "%"
+            tooltiptext = volume_now.left .. "% (" .. devicetype .. " " .. volume_now.index .. " \"" .. volume_now.device .. "\")"
             if volume_now.muted == "yes" then
-                widget:set_text("Mute")
-            else
-                widget:set_text(volume_now.left .. "%")
+                widgettext  = "Mute"
+                tooltiptext = "Mute/" .. tooltiptext
             end
+            widget:set_text(widgettext)
+            tooltip:set_text(tooltiptext)
         end
-        
+    
     pulseaudio.widget = wibox.widget.textbox()
-
+    pulseaudio.tooltip = awful.tooltip({ objects = { pulseaudio.widget } })
```

Comment:
Agreed with you in the first place, but on a second thought defining the tooltip here really makes sense though. Otherwise it becomes impossible to _easily_ update it's contents in the settings function, I guess. Also it doesn't actually hurt performance, does it?